Feature/9/van subscribe

mxmar opened the pull request:
PR for https://github.com/lukso-network/vanguard-consensus-engine/issues/9

blazejkrzak committed:
wtf

blazejkrzak committed:
Test fail on state not on the length

blazejkrzak committed:
Unknown boundary state error

blazejkrzak committed:
I am progressing further - unknown block

blazejkrzak committed:
Still unknown block in test

mxmar committed:
Merge remote-tracking branch 'origin/feature/GetProposerListForEpoch' into feature/9/van-subscribe

# Conflicts:
#	beacon-chain/rpc/beacon/assignments.go

mxmar committed:
Feature: 9 van subscribe
- added new package to the beacon-chain;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 van subscribe
- added bazel build files;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 van subscribe
- added bazel build files;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 van subscribe
- modified bazel build files;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 van subscribe
- added service registration in beacon chain node and changed default config;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 van subscribe
- refactored - bazel nogo feedback;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 van subscribe
- refactored - bazel nogo struct feedback;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 van subscribe
- refactored - working bazel test and build;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 van subscribe
- modified node bazel build file;

Signed-off-by: mxmar <macht4tech@gmail.com>

atif-konasl commented on beacon-chain/node/node.go:
```diff
@@ -658,6 +663,24 @@ func (b *BeaconNode) registerRPCService() error {
 }
+func (b *BeaconNode) registerSubscriberRPCService() error {
+	config := &subscriber.Config{
+		IPCPath:    cmd.DefaultIpcPath,
```
These endpoint values should be taken from flag.

mxmar replied:
You're right. I won't change it for now, because we're moving to the gRPC smoothly.

atif-konasl commented on beacon-chain/rpc/beacon/assignments.go:
```diff
@@ -134,42 +138,48 @@ func (bs *Server) ListValidatorAssignments(
-	}
-
-	nextEpochStratSlot, err := helpers.StartSlot(nextEpoch)
+func (bs *Server) GetProposerListForEpoch(
```
Why do we need this `GetProposerListForEpoch` or `NextEpochProposerList ` now? We do not call this api now.

blazejkrzak replied:
we want to fetch certain epoch, its not necessary `next`

atif-konasl replied:
Yeah, And we also do not need `GetProposerListForEpoch` this method too. This method used for grpc api over http.

mxmar replied:
It's now private for `bs - beacon server` and we'are using it inside `GetMinimalConsensusInfo`

atif-konasl commented on beacon-chain/rpc/beacon/assignments.go:
```diff
@@ -178,38 +188,46 @@ func (bs *Server) NextEpochProposerList(
+	return
+}
+
+func (bs *Server) GetMinimalConsensusInfo(
```
Why do we need this `GetMinimalConsensusInfo` now? We do not call this api now.

mxmar replied:
It's called by api backend of the subscriber rpc implementation: `consensusInfo, currentErr := beaconChain.GetMinimalConsensusInfo(ctx, blockEpoch)`

atif-konasl commented on beacon-chain/rpc/beacon/assignments_test.go:
```diff
@@ -393,3 +393,152 @@ func TestServer_ListAssignments_CanFilterPubkeysIndices_WithPagination(t *testin
 
 	assert.DeepEqual(t, wantedRes, res, "Did not receive wanted assignments")
 }
+
```
Same to here

mxmar replied:
We've refactor tests (new: `TestServer_MinimalConsensusSuite`)

atif-konasl commented on beacon-chain/rpc/beacon/assignments_test.go:
```diff
@@ -393,3 +393,152 @@ func TestServer_ListAssignments_CanFilterPubkeysIndices_WithPagination(t *testin
+	})
+}
+
+func TestServer_GetMinimalConsensusInfo(t *testing.T) {
```
Same to here

mxmar replied:
Solved. New test is `TestServer_MinimalConsensusSuite`

atif-konasl commented on beacon-chain/rpc/subscriber/api/events/backend_service_mock.go:
```diff
@@ -0,0 +1,38 @@
+	deadline = 5 * time.Minute
+)
+
+type MockBackend struct {
```
Here you implement `Backend` an interface but these methods are not defined in that `interface`.

mxmar replied:
Solved. We removed this mock and using another test file - `api_backend_test.go`

atif-konasl commented on beacon-chain/rpc/subscriber/api/events/handler.go:
```diff
@@ -0,0 +1,203 @@
+}
+
+// sendConsensusInfo
+func (es *EventSystem) sendConsensusInfo(f *subscription, ev *MinimalEpochConsensusInfo) {
```
This method does not need. When any subscriber is registered, it gets all the previous consensus info from `fromEpoch` to current epoch.

mxmar replied:
Solved. We removed custom event system and manged to use `stateNotifier.StateFeed().Send(&feed.Event)`

atif-konasl reviewed (commented):
Where will the consensus info event be triggered? I have not found any event send implementation.

blazejkrzak committed:
Changed defaults of the command

blazejkrzak committed:
Changed defaults of the command

blazejkrzak committed:
Bazel works

blazejkrzak committed:
Still having problems with assetions

blazejkrzak committed:
Fixed LISKOV princible breakage

blazejkrzak committed:
Added more debug

Commit:
Changed listener host

blazejkrzak committed:
Try to have less naive approach

blazejkrzak committed:
Merge branch 'feature/9/van-subscribe' of github.com:lukso-network/vanguard-consensus-engine into feature/9/van-subscribe

blazejkrzak committed:
Added time variation, bazel build will likely fail due to lack of `time` library. Tests were run only in go

mxmar committed:
Feature: 9 vanguard subscriber
- added logger for rpc debugging;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscriber
- fixed test and be logic;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscriber
- fixed test

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscriber
- debug test;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscriber
- debug test;

Signed-off-by: mxmar <macht4tech@gmail.com>

blazejkrzak committed:
Feature: 9 van subscribe
- GetMinimalConsensusInfo is now returning proper information for certain epoch

blazejkrzak committed:
Feature: 9 van subscribe
- GetMinimalConsensusInfoRange is working as expected

blazejkrzak committed:
Feature: 9 van subscribe
- NextEpochProposerList test is covering epoch0

blazejkrzak committed:
Feature: 9 van subscribe
- More description

blazejkrzak committed:
Feature: 9 van subscribe
- temp commit

blazejkrzak committed:
Feature: 9 van subscribe
- MinimalConsensusInfo should handle subscriptions
- TODO: write tests for `sendMinimalConsensusRange` `handleMinimalConsensusSubscription`

blazejkrzak committed:
Feature: 9 van subscribe
- Refactor part 1
- TODO: write tests for `sendMinimalConsensusRange` `handleMinimalConsensusSubscription`

blazejkrzak committed:
Feature: 9 van subscribe
- Refactor part 1 -> test are running
- TODO: write tests for `sendMinimalConsensusRange` `handleMinimalConsensusSubscription`

blazejkrzak committed:
Feature: 9 van subscribe
- Refactor part 2
- TODO: write tests for `sendMinimalConsensusRange` `handleMinimalConsensusSubscription`

mxmar committed:
Feature: 9 vanguard subscriber
- tests;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscriber
- tests;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscriber
- test for subscription works;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscriber
- debug tests;

Signed-off-by: mxmar <macht4tech@gmail.com>

blazejkrzak committed:
Feature: 9 van subscribe
- Test for subscription pass

blazejkrzak committed:
Feature: 9 van subscribe
- Removed tests that were doubled

blazejkrzak committed:
Feature: 9 van subscribe
- Bazel build should pass

mxmar committed:
Feature: 9 vanguard subscriber
- removed logger to file;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscriber
- added beaconServer to vanguard rpc;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscriber
- removed context with cancel;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscribe
- added deps to bazel build file;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscriber
- modified vanguard code;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Merge remote-tracking branch 'origin/feature/9/van-subscribe' into feature/9/van-subscribe

blazejkrzak committed:
Temp commit

blazejkrzak committed:
Merge branch 'feature/9/van-subscribe' of github.com:lukso-network/vanguard-consensus-engine into feature/9/van-subscribe

 Conflicts:
	beacon-chain/rpc/beacon/assignments.go

mxmar committed:
Feature: 9 vanguard subscriber
- added TestProposerAssignments;

Signed-off-by: mxmar <macht4tech@gmail.com>

blazejkrzak committed:
Assignments are having proper length and are encoded to 0x

blazejkrzak committed:
Merge branch 'feature/9/van-subscribe' of github.com:lukso-network/vanguard-consensus-engine into feature/9/van-subscribe

mxmar committed:
Feature: 9 vanguard subscribe
- debug context canceled;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscribe
- debug context canceled;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscribe
- debug context canceled;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscribe
- debug context canceled;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscriber
- added TODOs;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Merge remote-tracking branch 'origin/feature/9/van-subscribe' into feature/9/van-subscribe

mxmar committed:
Feature: 9 vanguard subscribe
- test modified - context;

Signed-off-by: mxmar <macht4tech@gmail.com>

blazejkrzak committed:
Feature: 9
- genesis epoch slot 0 is now `0x000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000`

blazejkrzak committed:
Merge branch 'feature/9/van-subscribe' of github.com:lukso-network/vanguard-consensus-engine into feature/9/van-subscribe

mxmar committed:
Feature: 9 vanguard subscriber
- added loging;

Signed-off-by: mxmar <macht4tech@gmail.com>

blazejkrzak committed:
Feature: 9
Panic in test

blazejkrzak committed:
Merge branch 'feature/9/van-subscribe' of github.com:lukso-network/vanguard-consensus-engine into feature/9/van-subscribe

mxmar committed:
Feature: 9 vanguard subscribe
- debugging;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscriber
- reverted debug logic;

Signed-off-by: mxmar <macht4tech@gmail.com>

blazejkrzak committed:
Feature: 9
NextEpochProposerList without tests. We should test it.

blazejkrzak committed:
Feature: 9
NextEpochProposerList pass tests

blazejkrzak committed:
Feature: 9
Change in NextEpochProposerList api

blazejkrzak committed:
Feature: 9
NextEpochProposerList attached when sending range of epochs

blazejkrzak committed:
Feature: 9
Tests should pass for subscriber

blazejkrzak committed:
Feature: 9
Tests should pass for assignments_test.go

blazejkrzak committed:
Feature: 9
Van subscribe could be tested live

blazejkrzak committed:
Feature: 9
Add additional debug to validator client

blazejkrzak committed:
Feature: 9
Possible flaw fix at GetMinimalConsensusInfo

blazejkrzak committed:
Feature: 9
Debug further

blazejkrzak committed:
Feature: 9
It looks like the unsorted problem should be fixed

blazejkrzak committed:
Feature: 9
It looks like the unsorted problem should be fixed

blazejkrzak committed:
Feature: 9
Early return when vanguard is not yet ready

blazejkrzak committed:
Feature: 9
There should be no major panic gaps

blazejkrzak committed:
Feature: 9
Added tryout to send future epoch in routine instead of 1 by 1

blazejkrzak committed:
Feature: 9
Fixed bazel build

mxmar committed:
Feature: 9 vanguard subscriber
- ticker should work

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscriber
- tests should work;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscriber
- test modified;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscriber
- test modified;
- debug ctx;

Signed-off-by: mxmar <macht4tech@gmail.com>

mxmar committed:
Feature: 9 vanguard subscriber
- test removed for now;

Signed-off-by: mxmar <macht4tech@gmail.com>
